internal/pkg/cli/task_run.go

```diff
@@ -0,0 +1,242 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/docker/dockerfile"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type runTaskVars struct {
+	*GlobalOpts
+	Num    uint8
+	CPU    uint16
+	Memory uint16
+
+	Image          string
+	DockerfilePath string
+
+	TaskRole string
+
+	SubnetID         string
+	SecurityGroupIDs []string
+	App              string
```

Comment:
We don't need this since it is already in `GlobalOpts`

Reply:
The `GlobalOpts`'s `appName` is a bit different than the `app` flag here. You can call `task run` under a workspace but specify a different app that you want to deploy the task to.

Reply:
I might not fully understand but why can't we you use `appName` in `GlobalOpts` to specify app other than the one associated with the local workspace? I remember we use it in other commands like `svc show`. If you are in a workspace that belongs to `appA` and call `copilot svc show --app appB` it shows the service in appB instead of appA.

Reply:
Under the hood `AppName()` looks for the value for `appName` first, which gets passed in value from flag. If no app flag set, then the value for `appName` is empty then `AppName()` will look for the app associated with the local workspace. But if there's no local workspace found then `AppName()` returns empty string. Normally when that happens we'll prompt users to choose from an existing application.

Reply:
Yes you are right! I will make the changes accordingly.